zephyr/lib/alloc.c

```diff
@@ -642,7 +661,7 @@ void *sof_heap_alloc(struct k_heap *heap, uint32_t flags, size_t bytes,
 
 void sof_heap_free(struct k_heap *heap, void *addr)
 {
-	if (heap && addr)
+	if (heap && addr && is_heap_pointer(heap, addr))
```

Comment:
This seems a bit complex, but this would indeed seem to be a needed check, also for SOF_MEM_FLAG_LARGE_BUFFER .